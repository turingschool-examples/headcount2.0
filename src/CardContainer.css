.cardContainer-grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(255px, 1fr));
  grid-gap: 15px;
  padding: 10px;
  margin: 10px;
  border-radius: 25px;
  border: cornflowerblue;
  align-items: center;
}

.cardContainerHeading {
  color: cornflowerblue; 
  text-align: center;
}