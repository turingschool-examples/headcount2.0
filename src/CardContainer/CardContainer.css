.card-container {
  display: flexbox;
  flex-wrap: wrap;
  border: black 1px solid;
  display: flex;
  flex-wrap: wrap;
  width: 84%;
  height: 500px;
  margin: 5px auto;
  justify-content: center;
  overflow-y: scroll;
}