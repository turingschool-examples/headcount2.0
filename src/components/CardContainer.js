import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';
import CompareCard from './CompareCard';

const CardContainer = ({handleData, handleCompare, handleCompareData, handleComparison}) => {
  const districtArray = Object.keys(handleData).map((districtObj) => <Card key={handleData[districtObj].location} {...handleData[districtObj]} handleCompare={handleCompare}/>);

  const compareArray = handleCompareData.map((compareObj, index) => {
    if (index === 2) {
      return (
      <CompareCard key={index} {...compareObj} />
      )
    }
    return (
    <Card key={compareObj.location} active={true} handleCompare={handleCompare} {...compareObj}/>
    )

  })

  return(
    <section className='cardsContainer'>
      <div className='compareCards'>
        {compareArray}
      </div>
      <div className='cardList'>
        {districtArray}
      </div>
    </section>
  )
}

CardContainer.propTypes = {
  handleData: PropTypes.object,
  handleCompare: PropTypes.func,
  handleCompareData: PropTypes.array,
}

export default CardContainer;
