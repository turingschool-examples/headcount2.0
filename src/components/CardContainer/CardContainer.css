/*CARDCONTAINER COMPONENT RELATED CSS GOES HERE*/
.CardContainer {
  grid-row: 3 / 4;
  grid-column: 1 / 2;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  padding-top: 15px;
}