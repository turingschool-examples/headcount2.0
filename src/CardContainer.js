import React, { Component } from 'react';
import Card from './Card';
import './styles/CardContainer.css';

const CardContainer = ({ schoolData }) => {
  const schoolNames = Object.keys(schoolData);

  const cards = schoolNames.map((schoolName, index) => {
    const cardData = schoolData[schoolName]
    return <Card cardData={cardData} key={index} />
  })
 return cards;
}

export default CardContainer;