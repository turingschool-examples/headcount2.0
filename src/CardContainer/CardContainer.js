import React from 'react';
import Card from '../Card/Card.js';
import PropTypes from 'prop-types';
import './CardContainer.css'

const CardContainer = ({districtsData}) => {
  // console.log(districtsData);
  
  const districtCards = Object.keys(districtsData).map((key, index) => {
    return (
      <Card key={index} district={districtsData[key]} />
    )
  });

  return(
    <div className="card-container">
      {districtCards}
    </div>
  )
}

CardContainer.propTypes = {
  districtsData: PropTypes.any.isRequired
}

export default CardContainer;

      