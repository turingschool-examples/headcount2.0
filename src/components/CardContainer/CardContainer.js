import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

import './CardContainer.css';

const CardContainer = props => {
  const { districts } = props;

  const districtData = districts.map(district => <Card {...district} key={district.location} />);

  return (
    <div>
      <h1>Card Container</h1>
      <div className="card-container">{districtData}</div>
    </div>
  );
};

CardContainer.propTypes = {
  districts: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default CardContainer;
