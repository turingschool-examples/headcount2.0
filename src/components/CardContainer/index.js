import React from 'react'
import PropTypes from 'prop-types'
import Card from '../Card'
import './CardContainer.css'

const CardContainer = ({districts, adjustComparisons}) => {
  const cards = Object.keys(districts).map(district => {
    const stats = districts[district].stats

    return <Card key={`${district}-district`}
                 location={district}
                 stats={stats}
                 adjustComparisons={adjustComparisons}
           />
  })

  return(
    <main className="CardContainer">
       { cards }
    </main>
  )
}

CardContainer.propTypes = {
  districts: PropTypes.object.isRequired
}

export default CardContainer