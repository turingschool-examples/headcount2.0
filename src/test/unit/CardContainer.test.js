import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from '../../CardContainer';
import { shallow, mount } from 'enzyme';

describe('CardContainer', () => {
  let cardContainer;

  beforeEach(() => {
    cardContainer = shallow(<CardContainer />)
  })

  it('matches snapshot', () => {

    expect(cardContainer).toMatchSnapshot()
  });
})