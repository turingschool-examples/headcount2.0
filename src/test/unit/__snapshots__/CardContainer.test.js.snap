// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches snapshot 1`] = `undefined`;
